apps/bmd/src/utils/tallies.ts

```diff
@@ -6,10 +8,114 @@ import {
   VotesDict,
   YesNoVote,
 } from '@votingworks/types'
-import { CandidateVoteTally, Tally, YesNoVoteTally } from '../config/types'
+import {
+  CandidateVoteTally,
+  MsEitherNeitherTally,
+  Tally,
+  YesNoVoteTally,
+} from '../config/types'
 import { computeTallyForEitherNeitherContests } from './eitherNeither'
 import { getSingleYesNoVote } from './votes'
 
+const combineCandidateTallies = (
+  tally1: CandidateVoteTally,
+  tally2: CandidateVoteTally
+): CandidateVoteTally => {
+  const candidates = []
+  assert(tally1.candidates.length === tally2.candidates.length)
+  for (let i = 0; i < tally1.candidates.length; i++) {
+    candidates.push(tally1.candidates[i] + tally2.candidates[i])
+  }
+  const writeInVotes = tally1.writeIns
+  for (const writeInVote of tally2.writeIns) {
+    const existingWriteInIdx = writeInVotes.findIndex(
+      (v) => v.name === writeInVote.name
+    )
+    if (existingWriteInIdx >= 0) {
+      writeInVotes[existingWriteInIdx].tally += writeInVote.tally
+    } else {
+      writeInVotes.push(writeInVote)
+    }
+  }
+  return {
+    candidates,
+    undervotes: tally1.undervotes + tally2.undervotes,
+    writeIns: writeInVotes,
+  }
+}
+
+const combineYesNoTallies = (
+  tally1: YesNoVoteTally,
+  tally2: YesNoVoteTally
+): YesNoVoteTally => {
+  return {
+    yes: tally1.yes + tally2.yes,
+    no: tally1.no + tally2.no,
+    undervotes: tally1.undervotes + tally2.undervotes,
+  }
+}
+
+const combineEitherNeitherTallies = (
+  tally1: MsEitherNeitherTally,
+  tally2: MsEitherNeitherTally
+): MsEitherNeitherTally => {
+  return {
+    eitherOption: tally1.eitherOption + tally2.eitherOption,
+    neitherOption: tally1.neitherOption + tally2.neitherOption,
+    eitherNeitherUndervotes:
+      tally1.eitherNeitherUndervotes + tally2.eitherNeitherUndervotes,
+    firstOption: tally1.firstOption + tally2.firstOption,
+    secondOption: tally1.secondOption + tally2.secondOption,
+    pickOneUndervotes: tally1.pickOneUndervotes + tally2.pickOneUndervotes,
+  }
+}
+
+export const combineTallies = (
+  election: Election,
+  tally1: Tally,
+  tally2: Tally
+): Tally => {
+  assert(
+    election.contests.length === tally1.length &&
+      tally1.length === tally2.length
```

Comment:
I suggest using two assertions using `assert.equal` to get more information in the event of a crash.